Lib/pydoc.py

```diff
@@ -1617,13 +1618,13 @@ def pipepager(text, cmd):
 def tempfilepager(text, cmd):
     """Page through text by invoking a program on a temporary file."""
     import tempfile
-    filename = tempfile.mktemp()
-    with open(filename, 'w', errors='backslashreplace') as file:
-        file.write(text)
     try:
-        os.system(cmd + ' "' + filename + '"')
+        with tempfile.NamedTemporaryFile('w', delete=False, errors='backslashreplace') as file:
+            file.write(text)
+            file.close()
+            subprocess.run([cmd, file.name])
     finally:
-        os.unlink(filename)
+        os.unlink(file.name)
```

Comment:
There's no guarantee that `file` is set here (there can be an exception in `NamedTemporaryFile`, `KeyboardInterrupt` or something else), it probably doesn't matter at this point anyway but it'd generate an unhelpful error message.

I'd stick with the default `delete=True` parameter to `NamedTemporaryFile` and instead of `file.close()` call `file.flush()` before running the pager and then let the context manager take care of closing and deleting the file.

Reply:
Nice! This has also been implemented.